---
title: "The mice 16s sequencing data from diet experiments"
output: html_document
date: "2024-07-18"
---

```{r}
library(tidyverse)
library(ggpubr)
library(janitor)
library(kableExtra)
library(knitr)
library(Maaslin2)
library(vegan)
library(ggrepel)
library(scales)
library(readxl)
```

```{r}
healthyall <- read_csv('../data/204_mice_diet_healthy_all.csv')

meta <- healthyall %>%
  select(-Taxon, -relab) %>% 
  distinct() 
```
# alpha diversity 

```{r}
alpha_healthy_days <-meta %>% 
  mutate(day = factor(day)) %>% 
  arrange(abx_treatment,desc(diet_treatment),  day) %>% 
  mutate(xvar = str_glue('{abx_treatment}__{diet_treatment}__{day}')) %>% 
  mutate(grp = str_glue('{abx_treatment}__{diet_treatment}')) %>% 
  mutate(grp = factor(grp, levels = c('PBS__vehicle','PBS__sucrose','biapenem__vehicle','biapenem__sucrose'))) %>%    
  ggboxplot(x = 'xvar', y = 'simpson_reciprocal',add = 'jitter', 
             xlab = '', ylab = 'alpha diversity(simpson reciprocal)',, add.params = list(alpha = 0.5, shape = 16),
            width = 0.6,color  = 'grp')+
   scale_color_manual(values = c('gray76','#ffbcdc','gray32','deeppink2')) +
  stat_compare_means(comparisons = list(
    c('biapenem__vehicle__3','biapenem__sucrose__3'), 
    #c('biapenem__vehicle__6','biapenem__sucrose__6'),
    c('PBS__vehicle__3','PBS__sucrose__3'),
    c('PBS__vehicle__3','biapenem__vehicle__3')
    ),
    label= "p.signif", method= 'wilcox.test',tip.length = 0.04,exact=T,correct=T) +
  #scale_x_discrete(labels=rep(c(0,3,6),4)) + 
  scale_y_sqrt() +
  theme_light() +
  theme(axis.text =  element_text(size = 10, angle = 45, hjust = 1),axis.title=element_text(size=10),legend.position = 'none',aspect.ratio=1/1.3)
alpha_healthy_days
```
sucrose-only doesn’t reduce diversity on day 3, compared with vehicle, but biapenem-only does.  However, sucrose doesn’t further reduce diversity beyond the biapenem effect on day 3. 

# beta diversity and composition

```{r}
# the enterococccus relab
g_entero <- healthyall %>% filter(str_detect(Taxon, 'Enterococc')) %>% 
  group_by(sampleid) %>% 
  summarise(enterococcus = sum(relab))

# calculate my own bc distance using the subsetted samples

relab_df <- healthyall %>% 
  select(Taxon, sampleid, relab) %>% 
  spread(key = 'Taxon', value = 'relab', fill = 0) %>% 
  column_to_rownames('sampleid')

dist_ <- vegdist(relab_df, method = 'bray')
eigen <- cmdscale(dist_, eig = T)$eig
percent_var <- signif(eigen/sum(eigen), 3)*100
bc <- cmdscale(dist_, k = 2)
beta_all <- bc %>%
  as.data.frame() %>%
  rownames_to_column('sampleid')  %>% 
  full_join(meta) %>% 
  inner_join(g_entero) %>% 
  mutate(treatment = str_glue('{abx_treatment}_{diet_treatment}')) %>% 
  mutate(treatment = factor(treatment, levels = c('PBS_vehicle','PBS_sucrose','biapenem_vehicle','biapenem_sucrose')))
```

```{r}
# color by abx        
beta_all %>% 
  mutate(experiment_no = factor(experiment_no)) %>% 
  ggscatter(x = 'V1', y = 'V2', color = 'treatment', size = 'enterococcus', alpha = 0.6) +
  facet_grid(day ~ experiment_no) +
  theme_light() +
  scale_color_manual(values = c('gray76','#ffbcdc','gray32','deeppink2')) +
  xlab(paste0("PC 1 [",percent_var[1],"%]")) +
  ylab(paste0("PC 2 [",percent_var[2],"%]")) +
  theme(aspect.ratio=1, legend.position = 'right') 

ggsave('../data/R01_healthy_mice_pcoa_enterococcus.jpg', width = 6, height = 7)
```

```{r}
# to find the stool sample weight of these mice samples
# thanks to Madhu sending over the tables!
dat1 <- read_excel('../data/Angel Rebuttal_Exp2_Healthy mice Diet_Fructose_Sucrose_Glucose_v2.xlsx') %>% 
  rename(sampleid  = ...14) %>% 
  filter(!is.na(sampleid)) %>% 
  filter(!is.na(Stool_weight_mg)) %>% 
  select(sampleid, Stool_weight_mg)

dat2 <- read_excel('../data/Angel Rebuttal_Exp2_Healthy mice Diet_Fructose_Sucrose_Glucose_v2.xlsx')%>% 
  rename(sampleid  = ...14) %>% 
  filter(!is.na(sampleid)) %>% 
  filter(!is.na(Stool_weight_mg)) %>% 
  select(sampleid, Stool_weight_mg)

all_weights <- bind_rows(dat1, dat2) %>% distinct()

meta_weight <- meta %>% left_join(all_weights ) %>% filter(!is.na(Stool_weight_mg)) %>% clean_names()

meta_weight %>% filter(!is.na(Stool_weight_mg)) %>% count(abx_treatment, diet_treatment)
meta  %>% count(abx_treatment, diet_treatment)
```

  
```{r}
# the absolute copy numbers of enterococcus and plot them in boxplots
healthy_abso <- read_csv('../data/absolute.abundance.csv') %>% 
  rename(sampleid = customer_label) %>% 
  select(sampleid, gene_copies_per_ul) %>% 
  inner_join(meta_weight %>% select(sampleid, stool_weight_mg)) %>% 
  mutate(gene_copies_per_gram = gene_copies_per_ul * 500 /(stool_weight_mg/1000)) %>%
  inner_join(healthyall %>% 
    filter(str_detect(Taxon, 'g__Enterococc')) %>% 
    select(sampleid, day,relab,  abx_treatment:diet_treatment) %>% 
      group_by(sampleid, day,abx_treatment, diet_treatment) %>% 
    summarise(relab = sum(relab)) ) %>% 
  mutate(abs_copy = gene_copies_per_gram * relab,
         log10_abs_copy = log10(abs_copy + 1)) %>% 
  mutate(day = factor(day)) %>% 
  arrange(abx_treatment,desc(diet_treatment),  day) %>% 
  mutate(xvar = str_glue('{abx_treatment}__{diet_treatment}__{day}')) %>% 
  mutate(grp = str_glue('{abx_treatment}__{diet_treatment}')) %>% 
  mutate(grp = factor(grp, levels = c('PBS__vehicle','PBS__sucrose','biapenem__vehicle','biapenem__sucrose'))) 
```


```{r}
# plotting the data in boxplot
abso_entero_healthy_days <- healthy_abso %>%    
  ggboxplot(x = 'xvar', y = 'log10_abs_copy',add = 'jitter', 
             xlab = '', ylab = 'log10(Enterococcus copy numbers)',, add.params = list(alpha = 0.5, shape = 16),
            width = 0.6,color  = 'grp')+
   scale_color_manual(values = c('gray76','#ffbcdc','gray32','deeppink2')) +
  stat_compare_means(comparisons = list(
    c('biapenem__vehicle__3','biapenem__sucrose__3'), 
    #c('biapenem__vehicle__6','biapenem__sucrose__6'),
    c('PBS__vehicle__3','PBS__sucrose__3'),
    c('PBS__vehicle__3','biapenem__vehicle__3')
    ),
    label= "p.signif", method= 'wilcox.test',tip.length = 0.04,exact=T,correct=T) +
  #scale_y_log10(breaks = c(1 %o% 10^(-1:8)),  labels = trans_format("log10", math_format(10^.x))) +
  #scale_x_discrete(labels=rep(c(0,3,6),4)) + 
  theme_light() +
  theme(axis.text.x =  element_text(size = 10, angle = 45, hjust = 1),axis.title=element_text(size=10),legend.position = 'none',aspect.ratio=1/1.3)
#abso_entero_healthy_days
  
# this data is looking really weird!!!!, so it doesn't have much to do with the stool weight, the gene copy number is already weird, a lesson for future sequencing of mice samples
```
```{r}
# maybe just plot the relative abundance for enterococcus
Entero_healthy <- healthyall %>% 
    filter(str_detect(Taxon, 'g__Enterococc')) %>% 
    select(sampleid, day,relab,  abx_treatment:diet_treatment) %>% 
      group_by(sampleid, day,abx_treatment, diet_treatment) %>% 
    summarise(relab = sum(relab)) %>% 
  mutate(day = factor(day)) %>% 
  arrange(abx_treatment,desc(diet_treatment),  day) %>% 
  mutate(xvar = str_glue('{abx_treatment}__{diet_treatment}__{day}')) %>% 
  mutate(grp = str_glue('{abx_treatment}__{diet_treatment}')) %>% 
  mutate(grp = factor(grp, levels = c('PBS__vehicle','PBS__sucrose','biapenem__vehicle','biapenem__sucrose'))) %>% 
  
  ggboxplot(x = 'xvar', y = 'relab',add = 'jitter', 
             xlab = '', ylab = 'Enterococcus relative abundance',, add.params = list(alpha = 0.5, shape = 16),
            width = 0.6,color  = 'grp')+
   scale_color_manual(values = c('gray76','#ffbcdc','gray32','deeppink2')) +
  stat_compare_means(comparisons = list(c('biapenem__vehicle__3','biapenem__sucrose__3'), c('biapenem__vehicle__6','biapenem__sucrose__6')),label= "p.signif", method= 'wilcox.test',tip.length = 0.04,exact=F) +
  #scale_x_discrete(labels=rep(c(0,3,6),4)) + 
  theme(axis.text =  element_text(size = 10, angle = 45, hjust = 1),axis.title=element_text(size=10),legend.position = 'none',aspect.ratio=1/1.3)
Entero_healthy
```


# bar plot 

```{r}
taxa_annot <- read_csv('../data/204_zymo_annotation_taxa_color.csv')

heathy_relab_df <- healthyall  %>% 
  separate(Taxon, into = c('asv','taxa'), sep = ':') %>% 
  select(-taxa) %>% 
  left_join(taxa_annot) %>% 
  # not sure why some asv doesn't have annotation, so make them "other"
  mutate(color = if_else(is.na(color), '#BBBBBB', color)) %>% 
  mutate(grp = str_glue('{abx_treatment}__{diet_treatment}')) %>% 
  # this velow relevel is important for the order of the group
  mutate(grp = factor(grp, levels = c('PBS__vehicle','PBS__sucrose','biapenem__vehicle','biapenem__sucrose')))
  

asv_color <- heathy_relab_df %>% 
  distinct(asv, color, .keep_all = T) %>% 
  arrange(phylum, class, order, family, genus) %>% 
  select(asv, color) %>%    
  arrange(color) %>% 
  deframe()

panels4 <- heathy_relab_df %>%  
  mutate(xvar = str_glue('{experiment_no}__{mouse_no}')) %>% 
  split(.$grp) %>% 
  imap(function(df, name_){
   plots <-  df %>% 
  ggplot( aes(x=xvar, y=relab, fill=asv) ) +
  scale_fill_manual(values = asv_color) +
  labs(x = '', y = 'Relative abundance', title = str_glue('{name_}')) +
  geom_bar(stat = "identity",position="fill",width = 1) +
  theme_classic() +
  facet_grid(day ~ ., scales = 'free') +
  theme(legend.position = 'none', axis.text.x = element_text(angle = 45, hjust = 1))
   return(plots)
  })
 
library(cowplot) 
h_bar4 <- plot_grid(plotlist =panels4, nrow = 2 )
h_bar4
ggsave('../data/R01_h_bar4.pdf', width = 8, height = 12)
ggsave('../data/R01_h_bar4.jpg', width = 8, height = 12)
```

# Maaslin2 asv

##  sugar effect on top of abx effect

```{r}
# to put all the healthy mice data in one single model
df_target <- meta  %>%
  select(-group, -tube_no, -investigator) %>%
  mutate(experiment_no = as.character(experiment_no),
         day = as.character(day)) %>% 
  mutate(treatment = str_glue('{abx_treatment}_{diet_treatment}'),
       treatment = factor(treatment, levels = c('biapenem_vehicle','biapenem_sucrose', 'PBS_vehicle','PBS_sucrose'))) 
```

```{r}
asv_target <- healthyall %>%
  inner_join(df_target %>% select(sampleid, treatment)) %>% 
  select(Taxon, sampleid, relab, treatment, day, experiment_no)

asv_target_sample_n <- asv_target %>% distinct(sampleid) %>% nrow

kept_asv <- asv_target %>%
  group_by(Taxon) %>%
  count(relab > 0.0001) %>%
  rename(criteria = names(.)[2]) %>%
  filter(criteria == 'TRUE') %>%
  mutate(perc = round(n/asv_target_sample_n*100, 0)) %>%
  filter(perc > 10) %>%
  pull(Taxon)

asv_relab_target_fil <-asv_target %>%
  filter( Taxon %in% kept_asv) %>%
  spread('Taxon','relab', fill = 0) %>% 
  mutate(day = factor(day),
         experiment_no = factor(experiment_no))
```

```{r}
# the setup for the Maaslin2
metadata <- asv_relab_target_fil %>% select(sampleid:experiment_no) %>% column_to_rownames('sampleid')
formula <- ~ treatment * day + experiment_no
design_matrix <- model.matrix(formula, data = metadata)
metadata <- cbind(metadata, design_matrix)

paste(colnames(metadata) , collapse = ", ")
```

```{r}
# Create a MaAsLin2 input feature data file (ASV table)
feature_data <- asv_relab_target_fil %>% 
  select(sampleid, starts_with('seq')) %>% 
  column_to_rownames('sampleid')

fit_res <- Maaslin2(
  input_data = feature_data, 
  input_metadata = metadata,
  output = str_glue("../data/R01_maaslin2_output_ALL_grp4"),  # Output directory
  fixed_effects = c('treatmentbiapenem_sucrose', 'treatmentPBS_vehicle', 'treatmentPBS_sucrose', 'day3', 'day6', 'experiment_no2', 'treatmentbiapenem_sucrose:day3', 'treatmentPBS_vehicle:day3', 'treatmentPBS_sucrose:day3', 'treatmentbiapenem_sucrose:day6', 'treatmentPBS_vehicle:day6', 'treatmentPBS_sucrose:day6'), 
  random_effects = NULL,  # No random effects in this case
  normalization = "TSS",  # Total sum scaling normalization
  transform = "LOG",  # Log transformation actually log2
  analysis_method = "LM",  # Linear model
  max_significance = 0.05,
  min_abundance = 0.0001,  # Minimum abundance threshold
  min_prevalence = 0.10,  # Minimum prevalence threshold,
  cores = 10
)
sig_res = read_tsv(str_glue('../data/R01_maaslin2_output_ALL_grp4/significant_results.tsv'))

```

### volcano d3

```{r}
set.seed(1)
coeff_upper <- 1
coeff_lower <- -1
# to use the volcano plot to highlight the results on day 3
all_res <- read_tsv('../data/R01_maaslin2_output_ALL_grp4/all_results.tsv') %>% 
  mutate(genus = str_extract(feature, 'g__.+s__'),
         species = str_extract(feature, 's__.+$'),
         genus = str_replace(genus, 'g__',''),
         species = str_replace(species, 's__',''),
         genus = str_replace(genus, '\\.s__',''),
         cleaned_name = str_glue('{genus} {species}'))

# A coefficient of 2 means an 2²-fold change (approximately a 4-fold increase) in relative abundance.
# A coefficient of 3 means an 2³-fold change (approximately a 8-fold increase) in relative abundance.
 
all_res %>%
  filter(metadata == 'treatmentbiapenem_sucrose:day3') %>% 
  mutate(sig_ = if_else(qval < 0.05 & coef > 1, 'pos', if_else(qval < 0.05 & coef < -1, 'neg', 'not'))) %>% 
  mutate(neglog10_q = -log10(qval)) %>%
  ggscatter(x = 'coef', y = 'neglog10_q', color = 'sig_', alpha = 0.5, shape = 16, 
            xlab = 'Maaslin2 model coefficient', ylab = '- log10(q value)',
            title = 'Differential abundance between day 0 and day 3\nin abx + sucrose VS abx + vehicle') +
  scale_color_manual(values = c('blue', 'black', 'red')) +
  geom_vline(xintercept =  0, color = 'black') + 
  geom_vline(xintercept =  c(coeff_upper, coeff_lower), color = 'gray', linetype = 'dashed') + 
  geom_hline(yintercept =  -log10(0.05), color = 'gray', linetype = 'dashed') + 
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef > coeff_upper, cleaned_name, "")), color = 'red') +
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef < coeff_lower, cleaned_name, "")), color = 'blue') +
  theme(aspect.ratio = 1/2, legend.position = 'none')

```

### volcano d6

```{r}
# asv at day 6
all_res %>%
  filter(metadata == 'treatmentbiapenem_sucrose:day6') %>% 
  mutate(sig_ = if_else(qval < 0.05 & coef > 1, 'pos', if_else(qval < 0.05 & coef < -1, 'neg', 'not'))) %>% 
  mutate(neglog10_q = -log10(qval)) %>%
  ggscatter(x = 'coef', y = 'neglog10_q', color = 'sig_', alpha = 0.5, shape = 16, 
            xlab = 'Maaslin2 model coefficient', ylab = '- log10(q value)',
            title = 'Differential abundance between day 0 and day 6\nin abx + sucrose VS abx + vehicle') +
  scale_color_manual(values = c('blue', 'black', 'red')) +
  geom_vline(xintercept =  0, color = 'black') + 
  geom_vline(xintercept =  c(coeff_upper, coeff_lower), color = 'gray', linetype = 'dashed') + 
  geom_hline(yintercept =  -log10(0.05), color = 'gray', linetype = 'dashed') + 
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef > coeff_upper, cleaned_name, "")), color = 'red') +
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef < coeff_lower, cleaned_name, "")), color = 'blue') +
  theme(aspect.ratio = 1/2, legend.position = 'none')

```
### a table for all q < 0.05 results

```{r}
sugar_effects_table <- all_res %>% 
  filter(metadata == 'treatmentbiapenem_sucrose:day6' | metadata == 'treatmentbiapenem_sucrose:day3') %>% 
  filter(qval < 0.05) %>% 
  select(-value) %>% 
  arrange(metadata)
```

## abx effect

```{r}
# use the PBS_vehicle as the reference level
df_target <- meta  %>%
  select(-group, -tube_no, -investigator) %>%
  mutate(experiment_no = as.character(experiment_no),
         day = as.character(day)) %>% 
  mutate(treatment = str_glue('{abx_treatment}_{diet_treatment}'),
       treatment = factor(treatment, levels = c( 'PBS_vehicle','PBS_sucrose', 'biapenem_vehicle','biapenem_sucrose'))) 

asv_target <- healthyall %>%
  inner_join(df_target %>% select(sampleid, treatment)) %>% 
  select(Taxon, sampleid, relab, treatment, day, experiment_no)

asv_target_sample_n <- asv_target %>% distinct(sampleid) %>% nrow

kept_asv <- asv_target %>%
  group_by(Taxon) %>%
  count(relab > 0.0001) %>%
  rename(criteria = names(.)[2]) %>%
  filter(criteria == 'TRUE') %>%
  mutate(perc = round(n/asv_target_sample_n*100, 0)) %>%
  filter(perc > 10) %>%
  pull(Taxon)

asv_relab_target_fil <-asv_target %>%
  filter( Taxon %in% kept_asv) %>%
  spread('Taxon','relab', fill = 0) %>% 
  mutate(day = factor(day),
         experiment_no = factor(experiment_no))
```

```{r}
# the setup for the Maaslin2
metadata <- asv_relab_target_fil %>% select(sampleid:experiment_no) %>% column_to_rownames('sampleid')
formula <- ~ treatment * day + experiment_no
design_matrix <- model.matrix(formula, data = metadata)
metadata <- cbind(metadata, design_matrix)

paste(colnames(metadata) , collapse = ", ")
```

```{r}
# Create a MaAsLin2 input feature data file (ASV table)
feature_data <- asv_relab_target_fil %>% 
  select(sampleid, starts_with('seq')) %>% 
  column_to_rownames('sampleid')

fit_res <- Maaslin2(
  input_data = feature_data, 
  input_metadata = metadata,
  output = str_glue("../data/R01_maaslin2_output_ALL_grp4_pbs_reference_abx_effects"),  # Output directory
  fixed_effects = c('treatmentPBS_sucrose', 'treatmentbiapenem_vehicle', 'treatmentbiapenem_sucrose', 'day3', 'day6', 'experiment_no2', 'treatmentPBS_sucrose:day3', 'treatmentbiapenem_vehicle:day3', 'treatmentbiapenem_sucrose:day3', 'treatmentPBS_sucrose:day6', 'treatmentbiapenem_vehicle:day6', 'treatmentbiapenem_sucrose:day6'), 
  random_effects = NULL,  # No random effects in this case
  normalization = "TSS",  # Total sum scaling normalization
  transform = "LOG",  # Log transformation actually log2
  analysis_method = "LM",  # Linear model
  max_significance = 0.05,
  min_abundance = 0.0001,  # Minimum abundance threshold
  min_prevalence = 0.10,  # Minimum prevalence threshold,
  cores = 10
)
sig_res = read_tsv(str_glue('../data/R01_maaslin2_output_ALL_grp4_pbs_reference_abx_effects/significant_results.tsv'))

```

### volcano d3

```{r}
# to use the volcano plot to highlight the results on day 3
all_res <- read_tsv('../data/R01_maaslin2_output_ALL_grp4_pbs_reference_abx_effects/all_results.tsv') %>% 
  mutate(genus = str_extract(feature, 'g__.+s__'),
         species = str_extract(feature, 's__.+$'),
         genus = str_replace(genus, 'g__',''),
         species = str_replace(species, 's__',''),
         genus = str_replace(genus, '\\.s__',''),
         cleaned_name = str_glue('{genus} {species}'))


all_res %>%
  filter(metadata == 'treatmentbiapenem_vehicle:day3') %>% 
  mutate(sig_ = if_else(qval < 0.05 & coef > 1, 'pos', if_else(qval < 0.05 & coef < -1, 'neg', 'not'))) %>% 
  mutate(neglog10_q = -log10(qval)) %>%
  ggscatter(x = 'coef', y = 'neglog10_q', color = 'sig_', alpha = 0.5, shape = 16, 
            xlab = 'Maaslin2 model coefficient', ylab = '- log10(q value)',
            title = 'Differential abundance between day 0 and day 3\nin DPBS + vehicle VS abx + vehicle') +
  scale_color_manual(values = c('blue', 'black', 'red')) +
  geom_vline(xintercept =  0, color = 'black') + 
  geom_vline(xintercept =  c(coeff_upper, coeff_lower), color = 'gray', linetype = 'dashed') + 
  geom_hline(yintercept =  -log10(0.05), color = 'gray', linetype = 'dashed') + 
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef > coeff_upper, feature, "")), color = 'red') +
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef < coeff_lower, feature, "")), color = 'blue') +
  theme(aspect.ratio = 1/2, legend.position = 'none')
```

### volcano d6  

```{r}
all_res %>% 
  filter(metadata == 'treatmentbiapenem_vehicle:day6') %>% 
  mutate(sig_ = if_else(qval < 0.05 & coef > 1, 'pos', if_else(qval < 0.05 & coef < -1, 'neg', 'not'))) %>% 
  mutate(neglog10_q = -log10(qval)) %>%
  ggscatter(x = 'coef', y = 'neglog10_q', 
            #color = 'sig_', 
            alpha = 0.5, shape = 16, 
            xlab = 'Maaslin2 model coefficient', ylab = '- log10(q value)',
            title = 'Differential abundance between day 0 and day 6\nin DPBS + vehicle VS abx + vehicle') +
  scale_color_manual(values = c('blue', 'black', 'red')) +
  geom_vline(xintercept =  0, color = 'black') + 
  geom_vline(xintercept =  c(coeff_upper, coeff_lower), color = 'gray', linetype = 'dashed') + 
  geom_hline(yintercept =  -log10(0.05), color = 'gray', linetype = 'dashed') + 
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef > coeff_upper, feature, "")), color = 'red') +
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef < coeff_lower, feature, "")), color = 'blue') +
  theme(aspect.ratio = 1/2, legend.position = 'none')
```

### a table for all q < 0.05 results

```{r}
abx_effects_table <- all_res %>% 
  filter(metadata == 'treatmentbiapenem_vehicle:day6' | metadata == 'treatmentbiapenem_vehicle:day3') %>% 
  filter(qval < 0.05) %>% 
  select(-value) %>% 
  arrange(metadata)
```




# Maaslin2 family


```{r}
# at the family level
asv_target_family <- asv_target %>%
  separate(Taxon, into = c('asv','taxa'), sep = ':') %>% 
  left_join(taxa_annot %>% select(asv, family)) %>% 
  group_by(sampleid,treatment, day, experiment_no, family ) %>% 
  summarise(family_relab = sum(relab)) %>% 
  ungroup() %>% 
  filter(!is.na(family))
asv_target_family %>% ungroup %>% distinct(family)
```


```{r}
asv_target_sample_n <- asv_target_family %>% distinct(sampleid) %>% nrow

kept_family <- asv_target_family %>%
  group_by(family) %>%
  count(family_relab > 0.001) %>%
  rename(criteria = names(.)[2]) %>%
  filter(criteria == 'TRUE') %>%
  mutate(perc = round(n/asv_target_sample_n*100, 0)) %>%
  filter(perc > 10) %>%
  pull(family)

remove_not_in_mice_gut <- c('Alcaligenaceae','Anaeroplasmataceae','Defluviitaleaceae')

family_relab_target_fil <- asv_target_family %>%
  filter( family %in% kept_family) %>%
  filter( !family %in% remove_not_in_mice_gut) %>%
  spread('family','family_relab', fill = 0) %>% 
  mutate(day = factor(day),
         experiment_no = factor(experiment_no))
```

```{r}
feature_data_family <- family_relab_target_fil %>% 
  select(-all_of(c('treatment', 'day', 'experiment_no'))) %>% 
  column_to_rownames('sampleid')

fit_res_family <- Maaslin2(
  input_data = feature_data_family, 
  input_metadata = metadata,
  output = str_glue("../data/R01_maaslin2_output_ALL_grp4_family"),  # Output directory
  fixed_effects = c('treatmentbiapenem_sucrose', 'treatmentPBS_vehicle', 'treatmentPBS_sucrose', 'day3', 'day6', 'experiment_no2', 'treatmentbiapenem_sucrose:day3', 'treatmentPBS_vehicle:day3', 'treatmentPBS_sucrose:day3', 'treatmentbiapenem_sucrose:day6', 'treatmentPBS_vehicle:day6', 'treatmentPBS_sucrose:day6'), 
  random_effects = NULL,  # No random effects in this case
  normalization = "TSS",  # Total sum scaling normalization
  transform = "LOG",  # Log transformation actually log2
  analysis_method = "LM",  # Linear model
  max_significance = 0.05,
  min_abundance = 0.0001,  # Minimum abundance threshold
  min_prevalence = 0.10,  # Minimum prevalence threshold,
  cores = 10
)
sig_res = read_tsv(str_glue('../data/R01_maaslin2_output_ALL_grp4_family/significant_results.tsv'))

```





```{r}
## volcano plot at the family level
# also make the volcano plot at the family level
# to use the volcano plot to highlight the results on day 3
all_res_family  <- read_tsv('../data/R01_maaslin2_output_ALL_grp4_family/all_results.tsv') 

# A coefficient of 2 means an 2²-fold change (approximately a 4-fold increase) in relative abundance.
# A coefficient of 3 means an 2³-fold change (approximately a 8-fold increase) in relative abundance.
 
all_res_family %>%
  filter(metadata == 'treatmentbiapenem_sucrose:day3') %>% 
  mutate(sig_ = if_else(qval < 0.05 & coef > 1, 'pos', if_else(qval < 0.05 & coef < -1, 'neg', 'not'))) %>% 
  mutate(neglog10_q = -log10(qval)) %>%
  ggscatter(x = 'coef', y = 'neglog10_q', color = 'sig_', alpha = 0.5, shape = 16, 
            xlab = 'Maaslin2 model coefficient', ylab = '- log10(q value)',
            title = 'Differential abundance between day 0 and day 3\nin abx + sucrose VS abx + vehicle\nat Family level') +
  scale_color_manual(values = c('blue', 'black', 'red')) +
  geom_vline(xintercept =  0, color = 'black') + 
  geom_vline(xintercept =  c(1, -1), color = 'gray', linetype = 'dashed') + 
  geom_hline(yintercept =  -log10(0.05), color = 'gray', linetype = 'dashed') + 
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef > 1, feature, "")), color = 'red') +
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef < -1, feature, "")), color = 'blue') +
  theme(aspect.ratio = 1, legend.position = 'none')

```

```{r}
famly_d6 <- all_res_family %>%
  filter(metadata == 'treatmentbiapenem_sucrose:day6') %>% 
  mutate(sig_ = if_else(qval < 0.05 & coef > 1, 'pos', if_else(qval < 0.05 & coef < -1, 'neg', 'not'))) %>% 
  mutate(neglog10_q = -log10(qval)) %>%
  ggscatter(x = 'coef', y = 'neglog10_q', color = 'sig_', alpha = 0.5, shape = 16, 
            xlab = 'Maaslin2 model coefficient', ylab = '- log10(q value)',
            title = 'Differential abundance between day 0 and day 6\nin abx + sucrose VS abx + vehicle\nat Family level') +
  scale_color_manual(values = c('blue', 'black', 'red')) +
  geom_vline(xintercept =  0, color = 'black') + 
  geom_vline(xintercept =  c(1, -1), color = 'gray', linetype = 'dashed') + 
  geom_hline(yintercept =  -log10(0.05), color = 'gray', linetype = 'dashed') + 
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef > 1, feature, "")), color = 'red') +
  geom_text_repel(aes(label = ifelse(qval < 0.05 & coef < -1, feature, "")), color = 'blue') +
  theme(aspect.ratio = 1, legend.position = 'none')
famly_d6
```

